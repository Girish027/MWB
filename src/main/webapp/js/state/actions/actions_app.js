
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
