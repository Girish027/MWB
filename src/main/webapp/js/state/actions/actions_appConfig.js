import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
