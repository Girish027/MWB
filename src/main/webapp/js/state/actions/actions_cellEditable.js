import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
