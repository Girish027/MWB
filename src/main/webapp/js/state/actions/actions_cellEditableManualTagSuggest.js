import store from 'state/configureStore';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as types from './types';

let controller;
export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE,
  stateKey,
});
export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_REMOVE,
  stateKey,
});

const getSuggestedTagsRequest = ({ stateKey, projectId, intent }) => ({
  type: types.GET_SUGGESTED_TAGS_REQUEST,
  stateKey,
  projectId,
  intent,
});

const getSuggestedTagsError = ({
  stateKey, projectId, intent, error,
}) => ({
  type: types.GET_SUGGESTED_TAGS_ERROR,
  stateKey,
  projectId,
  intent,
  error,
});

const getSuggestedTagsSuccess = ({
  stateKey, projectId, intent, suggestedTags,
}) => ({
  type: types.GET_SUGGESTED_TAGS_SUCCESS,
  stateKey,
  projectId,
  intent,
  suggestedTags,
});

export const getSuggestedTags = ({ stateKey, projectId, intent }) => (dispatch, getState) => new Promise((resolve, reject) => {
  if (controller) {
    controller.abort();
  }
  controller = new AbortController();
  const signal = controller.signal;
  dispatch(getSuggestedTagsRequest({ stateKey, projectId, intent }));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.suggestedTags, { projectId, intent, clientId });

  fetch(fetchUrl, {
    signal,
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const suggestedTags = json && json.length ? json : [];
      const intentIndex = suggestedTags.indexOf(intent);
      if (intentIndex >= 0) {
        suggestedTags.slice(intentIndex, 1);
      }
      dispatch(getSuggestedTagsSuccess({
        stateKey, projectId, intent, suggestedTags,
      }));
      resolve([...suggestedTags]);
    })
    .catch((error) => {
      if (error.name === 'AbortError') { return; }
      dispatch(getSuggestedTagsError({
        stateKey, projectId, intent, error,
      }));
      reject(error);
    });
});

export const cursorDown = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_DOWN,
  stateKey,
});

export const cursorUp = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_UP,
  stateKey,
});
