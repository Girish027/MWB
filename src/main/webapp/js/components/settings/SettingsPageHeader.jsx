import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import breadcrumbSegments from 'components/AppHeader/breadcrumbSegments';
import Constants from 'constants/Constants';
import {
  PageHeader,
  ToastNotification,
  Xmark,
  Button,
} from '@tfs/ui-components';
import { stopShowingServerMessage, changeRoute } from 'state/actions/actions_app.js';

export default class SettingsPageHeader extends Component {
  constructor(props) {
    super(props);

    this.props = props;
    this.types = Constants.NOTIFICATION.types;
  }

  closeNotification = () => {
    const { dispatch } = this.props;
    dispatch(stopShowingServerMessage());
  }

  getNotification = () => {
    const { app: { notificationType, serverMessage } = {} } = this.props;

    let notification = null;
    if (notificationType && serverMessage) {
      notification = (
        <ToastNotification
          type={notificationType}
          styleOverride={{
            top: '4px',
            width: '40%',
            display: 'inline-block',
            padding: '0px 8px',
            fontSize: '13px',
            fontWeight: 'normal',
            lineHeight: '20px',
          }}
        >
          {serverMessage}
          <Button
            type="flat"
            onClick={this.closeNotification}
            styleOverride={{
              marginLeft: '10px',
              ':hover': {
                boxShadow: 'none',
              },
              ':focus': {
                outline: 'none',
              },
            }}
          >
            <Xmark fill={notificationType === this.types.error ? '#c81919' : '#00ac86'} />
          </Button>
        </ToastNotification>
      );
    }
    return notification;
  }

  render() {
    const {
      selectedProject, selectedClient,
      match, history, itsURL,
    } = this.props;

    const breadcrumbs = breadcrumbSegments(match, {
      selectedClient,
      selectedProject,
    }, history, itsURL);

    return (
      <PageHeader
        title={Constants.PREFERENCES}
        breadcrumb={breadcrumbs}
      >
        {this.getNotification()}
      </PageHeader>
    );
  }
}

SettingsPageHeader.defaultProps = {
  history: {},
  selectedClient: {},
  selectedProject: {},
  match: {},
  itsURL: '',
  clientId: '',
  dispatch: () => {},
};

SettingsPageHeader.propTypes = {
  history: PropTypes.object,
  selectedClient: PropTypes.object,
  selectedProject: PropTypes.object,
  match: PropTypes.object,
  itsURL: PropTypes.string,
  dispatch: PropTypes.func,
};
