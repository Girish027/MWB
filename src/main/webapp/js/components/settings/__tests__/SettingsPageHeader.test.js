import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as appActions from 'state/actions/actions_app';

import SettingsPageHeader from 'components/settings/SettingsPageHeader';

describe('<SettingsPageHeader />', () => {
  let wrapper;
  let props = {
    dispatch: () => {},
    app: {
      notificationType: 'success',
      serverMessage: 'model built',
    },
  };

  const breadcrumb = [
    {
      value: 'Overview',
      label: 'Overview',
    },
  ];
  const testProps = {
    ...props,
    breadcrumb,
    title: 'Preferences',
  };

  beforeEach(() => {
    wrapper = mount(<SettingsPageHeader {...props} />);
  });

  describe('Creating an instance:', () => {
    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots:', () => {
    test('should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should render correctly on with breadcrumbs', () => {
      wrapper = shallow(<SettingsPageHeader {...testProps} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.dispatch = jest.fn();
      appActions.stopShowingServerMessage = jest.fn(() => 'stopShowingServerMessage');
    });

    describe('closeNotification:', () => {
      test('should close the notification on closeNotification click', () => {
        wrapper = shallow(<SettingsPageHeader {...props} />);
        wrapper.instance().closeNotification();
        expect(appActions.stopShowingServerMessage).toHaveBeenCalledWith();
        expect(props.dispatch).toHaveBeenCalledWith('stopShowingServerMessage');
      });
    });
  });
});
