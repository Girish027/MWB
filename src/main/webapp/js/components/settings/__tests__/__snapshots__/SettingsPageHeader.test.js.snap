// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsPageHeader /> Snapshots: should render correctly 1`] = `
<SettingsPageHeader
  app={
    Object {
      "notificationType": "success",
      "serverMessage": "model built",
    }
  }
  clientId=""
  dispatch={[Function]}
  history={Object {}}
  itsURL=""
  match={Object {}}
  selectedClient={Object {}}
  selectedProject={Object {}}
>
  <withTheme(PageHeader)
    breadcrumb={
      Array [
        Object {
          "data-route": undefined,
          "itsurl": "?clientid=&appid=",
          "label": "Overview",
          "onClick": [Function],
          "value": "Overview",
        },
      ]
    }
    title="Preferences"
  >
    <PageHeader
      breadcrumb={
        Array [
          Object {
            "data-route": undefined,
            "itsurl": "?clientid=&appid=",
            "label": "Overview",
            "onClick": [Function],
            "value": "Overview",
          },
        ]
      }
      onBackClick={[Function]}
      showBackButton={false}
      styleOverride={Object {}}
      subtitle=""
      tfsStyle={
        Object {
          "pageHeader": Object {
            "breadcrumb": Object {
              "color": "#9E9E9E",
              "display": "inline-block",
              "fontFamily": "Lato",
              "fontSize": "13px",
              "marginBottom": "5px",
            },
            "breadcrumbHover": Object {
              "color": "#003467",
              "cursor": "pointer",
            },
            "chevronIcon": Object {
              "cursor": "pointer",
              "display": "inline-flex",
              "marginRight": "3px",
            },
            "secondaryNavBar": Object {
              "alignItems": "center",
              "background": "#F6F7F8",
              "borderBottom": "1px solid #DADADD",
              "display": "flex",
              "flexGrow": 1,
              "height": "60px",
              "justifyContent": "space-between",
              "paddingLeft": "30px",
            },
            "statusNotificationContainer": Object {
              "alignItems": "center",
              "display": "inline-flex",
              "justifyContent": "center",
              "left": "0px",
              "padding": "20px 0",
              "pointerEvents": "none",
              "position": "absolute",
              "right": "0px",
              "top": "0px",
            },
            "statusNotificationContent": Object {
              "color": "#4A90E2",
              "display": "inline-block",
              "fontSize": "13px",
              "fontWeight": "normal",
              "lineHeight": "20px",
              "padding": "0px 0px 0px 8px",
            },
            "subtitle": Object {
              "color": "#9E9E9E",
              "fontFamily": "Lato",
              "fontSize": "13px",
              "marginTop": "5px",
            },
            "title": Object {
              "color": "#313F54",
              "fontFamily": "Lato",
              "fontSize": "20px",
              "fontWeight": "bold",
            },
          },
        }
      }
      title="Preferences"
    >
      <div
        data-radium={true}
        style={
          Object {
            "alignItems": "center",
            "background": "#F6F7F8",
            "borderBottom": "1px solid #DADADD",
            "display": "flex",
            "flexGrow": 1,
            "height": "60px",
            "justifyContent": "space-between",
            "paddingLeft": "30px",
          }
        }
      >
        <div
          data-radium={true}
          key=".0"
        >
          <div
            data-radium={true}
            key=".0"
            style={Object {}}
          >
            <withTheme(Breadcrumb)
              items={
                Array [
                  Object {
                    "data-route": undefined,
                    "itsurl": "?clientid=&appid=",
                    "label": "Overview",
                    "onClick": [Function],
                    "value": "Overview",
                  },
                ]
              }
              styleOverride={Object {}}
            >
              <Breadcrumb
                items={
                  Array [
                    Object {
                      "data-route": undefined,
                      "itsurl": "?clientid=&appid=",
                      "label": "Overview",
                      "onClick": [Function],
                      "value": "Overview",
                    },
                  ]
                }
                onKeyDown={[Function]}
                styleOverride={Object {}}
                tfsStyle={
                  Object {
                    "pageHeader": Object {
                      "breadcrumb": Object {
                        "color": "#9E9E9E",
                        "display": "inline-block",
                        "fontFamily": "Lato",
                        "fontSize": "13px",
                        "marginBottom": "5px",
                      },
                      "breadcrumbHover": Object {
                        "color": "#003467",
                        "cursor": "pointer",
                      },
                      "chevronIcon": Object {
                        "cursor": "pointer",
                        "display": "inline-flex",
                        "marginRight": "3px",
                      },
                      "secondaryNavBar": Object {
                        "alignItems": "center",
                        "background": "#F6F7F8",
                        "borderBottom": "1px solid #DADADD",
                        "display": "flex",
                        "flexGrow": 1,
                        "height": "60px",
                        "justifyContent": "space-between",
                        "paddingLeft": "30px",
                      },
                      "statusNotificationContainer": Object {
                        "alignItems": "center",
                        "display": "inline-flex",
                        "justifyContent": "center",
                        "left": "0px",
                        "padding": "20px 0",
                        "pointerEvents": "none",
                        "position": "absolute",
                        "right": "0px",
                        "top": "0px",
                      },
                      "statusNotificationContent": Object {
                        "color": "#4A90E2",
                        "display": "inline-block",
                        "fontSize": "13px",
                        "fontWeight": "normal",
                        "lineHeight": "20px",
                        "padding": "0px 0px 0px 8px",
                      },
                      "subtitle": Object {
                        "color": "#9E9E9E",
                        "fontFamily": "Lato",
                        "fontSize": "13px",
                        "marginTop": "5px",
                      },
                      "title": Object {
                        "color": "#313F54",
                        "fontFamily": "Lato",
                        "fontSize": "20px",
                        "fontWeight": "bold",
                      },
                    },
                  }
                }
              >
                <div
                  data-radium={true}
                >
                  <div
                    data-radium={true}
                    key=".$Overview"
                    style={
                      Object {
                        "color": "#9E9E9E",
                        "display": "inline-block",
                      }
                    }
                  >
                    <div
                      data-radium={true}
                      key=".$Overview"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      role="presentation"
                      style={
                        Object {
                          "color": "#9E9E9E",
                          "display": "inline-block",
                          "fontFamily": "Lato",
                          "fontSize": "13px",
                          "marginBottom": "5px",
                        }
                      }
                    >
                      Overview
                    </div>
                     
                    /Â 
                  </div>
                </div>
              </Breadcrumb>
            </withTheme(Breadcrumb)>
          </div>
          <div
            data-radium={true}
            key=".1"
            style={
              Object {
                "color": "#313F54",
                "fontFamily": "Lato",
                "fontSize": "20px",
                "fontWeight": "bold",
              }
            }
          >
            <withTheme(Label)
              onBackClick={[Function]}
              showBackButton={false}
              title="Preferences"
            >
              <Label
                onBackClick={[Function]}
                showBackButton={false}
                styleOverride={Object {}}
                tfsStyle={
                  Object {
                    "pageHeader": Object {
                      "breadcrumb": Object {
                        "color": "#9E9E9E",
                        "display": "inline-block",
                        "fontFamily": "Lato",
                        "fontSize": "13px",
                        "marginBottom": "5px",
                      },
                      "breadcrumbHover": Object {
                        "color": "#003467",
                        "cursor": "pointer",
                      },
                      "chevronIcon": Object {
                        "cursor": "pointer",
                        "display": "inline-flex",
                        "marginRight": "3px",
                      },
                      "secondaryNavBar": Object {
                        "alignItems": "center",
                        "background": "#F6F7F8",
                        "borderBottom": "1px solid #DADADD",
                        "display": "flex",
                        "flexGrow": 1,
                        "height": "60px",
                        "justifyContent": "space-between",
                        "paddingLeft": "30px",
                      },
                      "statusNotificationContainer": Object {
                        "alignItems": "center",
                        "display": "inline-flex",
                        "justifyContent": "center",
                        "left": "0px",
                        "padding": "20px 0",
                        "pointerEvents": "none",
                        "position": "absolute",
                        "right": "0px",
                        "top": "0px",
                      },
                      "statusNotificationContent": Object {
                        "color": "#4A90E2",
                        "display": "inline-block",
                        "fontSize": "13px",
                        "fontWeight": "normal",
                        "lineHeight": "20px",
                        "padding": "0px 0px 0px 8px",
                      },
                      "subtitle": Object {
                        "color": "#9E9E9E",
                        "fontFamily": "Lato",
                        "fontSize": "13px",
                        "marginTop": "5px",
                      },
                      "title": Object {
                        "color": "#313F54",
                        "fontFamily": "Lato",
                        "fontSize": "20px",
                        "fontWeight": "bold",
                      },
                    },
                  }
                }
                title="Preferences"
              >
                <div
                  data-radium={true}
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                    }
                  }
                >
                  Preferences
                </div>
              </Label>
            </withTheme(Label)>
          </div>
        </div>
        <withTheme(ToastNotification)
          key=".1"
          styleOverride={
            Object {
              "display": "inline-block",
              "fontSize": "13px",
              "fontWeight": "normal",
              "lineHeight": "20px",
              "padding": "0px 8px",
              "top": "4px",
              "width": "40%",
            }
          }
          type="success"
        >
          <ToastNotification
            styleOverride={
              Object {
                "display": "inline-block",
                "fontSize": "13px",
                "fontWeight": "normal",
                "lineHeight": "20px",
                "padding": "0px 8px",
                "top": "4px",
                "width": "40%",
              }
            }
            tfsStyle={
              Object {
                "toastNotification": Object {
                  "container": Object {
                    "borderRadius": "5px",
                    "fontWeight": "bold",
                    "left": "0px",
                    "margin": "0 auto",
                    "padding": "10px",
                    "position": "absolute",
                    "right": "0px",
                    "textAlign": "center",
                    "top": "50px",
                    "width": "40%",
                  },
                  "default": Object {
                    "backgroundColor": "#4A90E214",
                    "border": "1px solid #4A90E2",
                    "color": "#4A90E2",
                  },
                  "error": Object {
                    "backgroundColor": "#C8191914",
                    "border": "1px solid #C81919",
                    "color": "#C81919",
                  },
                  "icon": Object {
                    "marginTop": "-5px",
                    "verticalAlign": "middle",
                  },
                  "success": Object {
                    "backgroundColor": "#00AC8614",
                    "border": "1px solid #00AC86",
                    "color": "#00AC86",
                  },
                },
              }
            }
            type="success"
          >
            <div
              data-radium={true}
              style={
                Object {
                  "backgroundColor": "#00AC8614",
                  "border": "1px solid #00AC86",
                  "borderRadius": "5px",
                  "color": "#00AC86",
                  "display": "inline-block",
                  "fontSize": "13px",
                  "fontWeight": "normal",
                  "left": "0px",
                  "lineHeight": "20px",
                  "margin": "0 auto",
                  "padding": "0px 8px",
                  "position": "absolute",
                  "right": "0px",
                  "textAlign": "center",
                  "top": "4px",
                  "width": "40%",
                }
              }
            >
              <SuccessIcon
                fill="#00ac86"
                height="16px"
                key=".0"
                styleOverride={Object {}}
                width="16px"
              >
                <svg
                  height="16px"
                  style={Object {}}
                  viewBox="0 0 16 16"
                  width="16px"
                >
                  <path
                    d="M29,22a8,8,0,1,0,8,8A8,8,0,0,0,29,22ZM27.5,34.1,24,30.6l1.2-1.2,2.3,2.3,5.3-5.3L34,27.6Z"
                    fill="#00ac86"
                    transform="translate(-21 -22)"
                  />
                </svg>
              </SuccessIcon>
              <div
                data-radium={true}
                key=".1"
                role="presentation"
                style={
                  Object {
                    "display": "inline-block",
                    "marginLeft": "10px",
                  }
                }
              >
                model built
                <withTheme(Button)
                  key=".$.1"
                  onClick={[Function]}
                  styleOverride={
                    Object {
                      ":focus": Object {
                        "outline": "none",
                      },
                      ":hover": Object {
                        "boxShadow": "none",
                      },
                      "marginLeft": "10px",
                    }
                  }
                  type="flat"
                >
                  <Button
                    disabled={false}
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    styleOverride={
                      Object {
                        ":focus": Object {
                          "outline": "none",
                        },
                        ":hover": Object {
                          "boxShadow": "none",
                        },
                        "marginLeft": "10px",
                      }
                    }
                    tabIndex={0}
                    tfsStyle={
                      Object {
                        "button": Object {
                          "base": Object {
                            ":focus": Object {
                              "outline": "2px solid #009CDE",
                            },
                            "border": "1px solid transparent",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "fontSize": "13px",
                            "fontWeight": 600,
                            "minHeight": "35px",
                            "minWidth": "100px",
                            "outlineOffset": "2px",
                            "paddingLeft": "20px",
                            "paddingRight": "20px",
                            "wordBreak": "normal",
                          },
                          "disabled": Object {
                            "cursor": "not-allowed",
                            "flat": Object {
                              "color": "#666666",
                            },
                            "primary": Object {
                              "backgroundColor": "#E0E0E0",
                              "color": "#666666",
                            },
                            "secondary": Object {
                              "border": "2px solid #E0E0E0",
                              "color": "#666666",
                            },
                          },
                          "flat": Object {
                            ":active": Object {
                              "color": "#004C97",
                            },
                            ":hover": Object {
                              "boxShadow": "0px 2px 5px 0px #DBDBDB",
                              "color": "#336FAB",
                            },
                            "backgroundColor": "transparent",
                            "color": "#000000",
                            "minWidth": "auto",
                            "paddingLeft": "10px",
                            "paddingRight": "10px",
                          },
                          "primary": Object {
                            ":active": Object {
                              "backgroundColor": "#336FAB",
                              "boxShadow": "inset 0px 0px 3px 0px #004C97",
                              "color": "#FFFFFF",
                            },
                            ":hover": Object {
                              "backgroundColor": "#336FAB",
                              "boxShadow": "2px 3px 10px 0px #DBDBDB",
                              "color": "#FFFFFF",
                            },
                            "backgroundColor": "#004C97",
                            "color": "#FFFFFF",
                          },
                          "secondary": Object {
                            ":active": Object {
                              "border": "2px solid #336FAB",
                              "boxShadow": "inset 0px 0px 3px 0px #336FAB",
                              "color": "#336FAB",
                            },
                            ":hover": Object {
                              "border": "2px solid #336FAB",
                              "boxShadow": "0px 2px 5px 0px #DBDBDB",
                              "color": "#336FAB",
                            },
                            "backgroundColor": "#FFFFFF",
                            "border": "2px solid #004C97",
                            "color": "#004C97",
                          },
                        },
                      }
                    }
                    type="flat"
                  >
                    <button
                      data-radium={true}
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "border": "1px solid transparent",
                          "borderRadius": "4px",
                          "color": "#000000",
                          "cursor": "pointer",
                          "fontSize": "13px",
                          "fontWeight": 600,
                          "marginLeft": "10px",
                          "minHeight": "35px",
                          "minWidth": "auto",
                          "outlineOffset": "2px",
                          "paddingLeft": "10px",
                          "paddingRight": "10px",
                          "wordBreak": "normal",
                        }
                      }
                      tabIndex={0}
                      type="button"
                    >
                      <Xmark
                        fill="#00ac86"
                        height="1em"
                        width="1em"
                      >
                        <svg
                          height="1em"
                          viewBox="0 0 24 24"
                          width="1em"
                        >
                          <path
                            d="M.421 20.58a1.442 1.442 0 0 0 2.038 2.04l8.891-8.89a.242.242 0 0 1 .34 0l8.891 8.89a1.445 1.445 0 0 0 2.41-.64 1.46 1.46 0 0 0-.372-1.4l-8.891-8.89a.257.257 0 0 1-.071-.17.239.239 0 0 1 .071-.17l8.892-8.89A1.442 1.442 0 0 0 20.581.42L11.69 9.31a.242.242 0 0 1-.34 0L2.459.42A1.442 1.442 0 0 0 .421 2.46l8.891 8.89a.239.239 0 0 1 .071.17.257.257 0 0 1-.071.17z"
                            data-name="remove"
                            fill="#00ac86"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Xmark>
                    </button>
                  </Button>
                </withTheme(Button)>
              </div>
            </div>
          </ToastNotification>
        </withTheme(ToastNotification)>
      </div>
    </PageHeader>
  </withTheme(PageHeader)>
</SettingsPageHeader>
`;

exports[`<SettingsPageHeader /> Snapshots: should render correctly on with breadcrumbs 1`] = `
<withTheme(PageHeader)
  breadcrumb={
    Array [
      Object {
        "data-route": undefined,
        "itsurl": "?clientid=&appid=",
        "label": "Overview",
        "onClick": [Function],
        "value": "Overview",
      },
    ]
  }
  title="Preferences"
>
  <withTheme(ToastNotification)
    styleOverride={
      Object {
        "display": "inline-block",
        "fontSize": "13px",
        "fontWeight": "normal",
        "lineHeight": "20px",
        "padding": "0px 8px",
        "top": "4px",
        "width": "40%",
      }
    }
    type="success"
  >
    model built
    <withTheme(Button)
      onClick={[Function]}
      styleOverride={
        Object {
          ":focus": Object {
            "outline": "none",
          },
          ":hover": Object {
            "boxShadow": "none",
          },
          "marginLeft": "10px",
        }
      }
      type="flat"
    >
      <Xmark
        fill="#00ac86"
        height="1em"
        width="1em"
      />
    </withTheme(Button)>
  </withTheme(ToastNotification)>
</withTheme(PageHeader)>
`;
